code/game/machinery/spiderbot_spawner.dm

```diff
@@ -0,0 +1,224 @@
+/obj/machinery/spiderbot_fabricator
+    name = "\improper spiderbot fabricator"
+    desc = "A large pad sunk into the ground that brains with legs occasionally crawl out of."
+    icon = 'icons/obj/robotics.dmi'
+    icon_state = "mommispawner-idle"
+    density = TRUE
+    anchored = TRUE
+    machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK
+    var/building = FALSE
+    var/metal = 0
+    var/max_metal = 100
+    var/const/metalPerSpiderbot = 10
+    use_power = MACHINE_POWER_USE_IDLE
+    idle_power_usage = 20
+    active_power_usage = 5000
+    var/recharge_time = 120 SECONDS
+    var/recharge_time_reduction = 0
+    var/last_built_time = -120 SECONDS
+    var/build_time = 20 SECONDS
+    var/build_time_reduction = 0
+
+/obj/machinery/spiderbot_fabricator/proc/get_recharge_time()
+    return max(recharge_time - recharge_time_reduction, 0)
+
+/obj/machinery/spiderbot_fabricator/proc/get_build_time()
+    return max(build_time - build_time_reduction, 0)
+
+/obj/machinery/spiderbot_fabricator/New()
+    . = ..()
+    component_parts = newlist(
+        /obj/item/weapon/circuitboard/spiderbot_fabricator,
+        /obj/item/weapon/stock_parts/matter_bin,
+        /obj/item/weapon/stock_parts/manipulator,
+        /obj/item/weapon/stock_parts/manipulator,
+        /obj/item/weapon/stock_parts/micro_laser)
+    RefreshParts()
+
+//parent already calls spillContents for us
+/obj/machinery/spiderbot_fabricator/crowbarDestroy(mob/user, obj/item/tool/crowbar/I)
+    if(building)
+        to_chat(user, "<span class='warning'>You cannot disassemble \the [src] while it is building something.</span>")
+        return FALSE
+    return ..()
```

Comment:
This is fine but you still should do some due diligence in managing refs and contents if this machine is destroyed via bomb or singulo